File: core/src/main/java/com/larryhsiao/nyx/core/jots/WrappedJot.java
package com.larryhsiao.nyx.core.jots;

/**
 * Wrap object of Jot
 */
public class WrappedJot implements Jot {
    private final Jot jot;

    public WrappedJot(Jot jot) {
        this.jot = jot;
    }

    @Override
    public long id() {
        return jot.id();
    }

    @Override
    public String content() {
        return jot.content();
    }

    @Override
    public long createdTime() {
        return jot.createdTime();
    }

    @Override
    public String mood() {
        return jot.mood();
    }

    @Override
    public double[] location() {
        return jot.location();
    }

    @Override
    public int version() {
        return jot.version();
    }

    @Override
    public boolean deleted() {
        return jot.deleted();
    }
}

File: core/src/main/java/com/larryhsiao/nyx/core/jots/WrappedJot.kt
package com.larryhsiao.nyx.core.jots

/**
 * Wrap object of Jot
 */
open class WrappedJot(private val jot: Jot) : Jot {
    override fun id(): Long {
        return jot.id()
    }

    override fun title(): String {
        return jot.title()
    }

    override fun content(): String {
        return jot.content()
    }

    override fun createdTime(): Long {
        return jot.createdTime()
    }

    override fun mood(): String {
        return jot.mood()
    }

    override fun location(): DoubleArray {
        return jot.location()
    }

    override fun version(): Int {
        return jot.version()
    }

    override fun deleted(): Boolean {
        return jot.deleted()
    }

    override fun privateLock(): Boolean {
        return jot.privateLock()
    }
}